### DISCUSSION TITLE
Returning a custom nack

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#4642] [Q&A]
We are trying to do some testing with another system and want to make mirth return a nack on every message. Could someone explain how to do that and what the source response drop-down should be?

I tried the below but get warnings that the method is deprecated (The ACKGenerator constructor is now deprecated and will soon be removed.).  And, when you look at the response it still shows an ACK being returned.

`var serializer = SerializerFactory.getSerializer('HL7V2');
var ack = new XML(serializer.toXML(new ACKGenerator().generateAckResponse(connectorMessage.getRawData(),'AE','There was an error')));
ack.MSA['MSA.2']['MSA.2.1'] = 'AE';
 ack.MSA['MSA.2']['MSA.2.1'] = '';
 ack.MSA['MSA.3']['MSA.3.1'] = 'Test error';
return serializer.fromXML(ack);`

**Comment by tonygermano** [comment]
I can't tell exactly what you're trying to do by resetting the values multiple times. I think you can do this without any javascript code.

I think in the HL7v2 data type properties, you just need to set the Successful ACK code to `AE` and the Successful ACK Message to whatever you want to send. Then you should be able to use the auto-generated responses.
